cpp/include/cudf/wrappers/timestamps.hpp

```diff
@@ -37,12 +37,23 @@ using time_point = simt::std::chrono::time_point<simt::std::chrono::system_clock
 template <class Duration>
 struct timestamp : time_point<Duration> {
   constexpr timestamp() : time_point<Duration>(Duration()){};
-  constexpr timestamp(Duration d) : time_point<Duration>(d){};
+
+  // Implicitly convert a Duration to a timestamp
+  constexpr timestamp(Duration const& d) : time_point<Duration>(d){};
```

Comment:
I don't think this is something we should allow. Implicit conversions like this lead to surprising and unintended behavior.

Reply:
recommend making this `explicit` constructor. https://en.cppreference.com/w/cpp/chrono/time_point/time_point

Reply:
i did not change this other than to add a comment and to take the arg as a reference. i thought there was an original intent to relax the explicitness of the constructor that takes in a duration (as the std chrono version requires this to be explicit) and thought this was plausibly one of the reasons why this type was required and thus added the documentation to show where it digresses from `std::chrono::timepoint` constructor

Reply:
i have added a comment to #5359 to see if a couple of constructors can be inherited from the chrono `time_point` type